### PR TITLE
Provider assets

### PR DESCRIPTION
**Pull request by aleksrutins**
This pull request adds static assets support for providers. It uses a new `static_assets` method in `Provider` that returns a `HashMap` with filenames as keys and file contents as values. It puts files into `/assets` in the Docker container, and you can get the path to an asset by calling `app.asset_path(name: &str)`.

See #158

### CODE REVIEW

File: src/nixpacks/mod.rs [hunk 1]
```diff
@@ -381,6 +396,19 @@ impl<'a> AppBuilder<'a> {
         File::create(dockerfile_path.clone()).context("Creating Dockerfile file")?;
         fs::write(dockerfile_path, dockerfile).context("Writing Dockerfile")?;
 
+        let static_assets_path = PathBuf::from(dest).join(PathBuf::from("assets"));
+        fs::create_dir_all(&static_assets_path).context("Creating static assets folder")?;
```
Comment (coffee-cup) [inline src/nixpacks/mod.rs:400]: If there are no assets I don't think the `assets/` directory should be created

File: src/providers/mod.rs [hunk 2]
```diff
@@ -13,6 +15,8 @@ pub mod node;
 pub mod python;
 pub mod rust;
 
+pub type StaticAssets = HashMap<String, String>;
```
Comment (coffee-cup) [inline src/providers/mod.rs:18]: nit: Move this type out the providers module. Maybe to `app.rs`?

**Review by coffee-cup — COMMENTED** [review]
This is awesome! A few small comments.

Also, this doesn't seem to work with sub directories (i.e. specifying `dir/hello.txt` as the filename). It errors when attempting to copy the files. I think we should support this case or show a better error than

image

**Comment by aleksrutins** [comment]
Should all hopefully be resolved now.

**Review by coffee-cup — APPROVED** [review]
Thanks :D

**Pull request merged**